src/pages/login/login.ts

```diff
@@ -0,0 +1,129 @@
+import input from 'components/input'
+import Block from '../../core/Block'
+import { validateForm, ValidateType } from '../../helpers/validateForm'
+import { signUpPage } from '../../index'
+
+export class LoginPage extends Block {
+	constructor() {
+		super()
+
+		this.setProps({
+			error: '',
+			loginValue: '',
+			passwordValue: '',
+			onInput: e => {
+				this.refs.loginInputRef.refs.errorRef.setProps({ text: '' })
+				const inputEl = e.target as HTMLInputElement
+				if (this.element?.querySelector('input[name="login"]') === inputEl) {
+					const errorMessage = validateForm([
+						{ type: ValidateType.Login, value: inputEl.value },
+						//{ type: ValidateType.Password, value: inputEl.value },
+					])
+					this.refs.loginInputRef.refs.errorRef.setProps({
+						text: errorMessage,
+					})
+				}
+				if (this.element?.querySelector('input[name="password"]') === inputEl) {
+					const errorMessagePass = validateForm([
+						{ type: ValidateType.Password, value: inputEl.value },
+						//{ type: ValidateType.Password, value: inputEl.value },
+					])
+					this.refs.passwordInputRef.refs.errorRef.setProps({
+						text: errorMessagePass,
+					})
+				}
+			},
+			onBlur: () => console.log('blur'),
+			onFocus: () => {},
+			onSubmit: () => {
+				const btnSignIn = this.element?.querySelector('form .sign-btn')
+				event?.preventDefault()
+				//btnSignIn.preventDefault()
+				const loginEl = this.element?.querySelector(
+					'input[name="login"]'
+				) as HTMLInputElement
+				const passwordEl = this.element?.querySelector(
+					'input[name="password"]'
+				) as HTMLInputElement
```

Comment:
Нужно стараться избегать явного приведения типов там, где этого избежать можно. В данном случае можно написать так:
```suggestion
				const passwordEl = this.element?.querySelector<HTMLInputElement>(
					'input[name="password"]')
```

Reply:
изменил